### ISSUE TITLE
Incorrect permissions on /nix/var/nix/profiles/per-user

### ISSUE THREAD

**Issue by jmgilman** [nlewo/nix2container#49]
I'm creating a container with nix initialized. I mounted a local flake to it and ran `nix flake show` as a test but it failed with the following error:

> error: could not set permissions on '/nix/var/nix/profiles/per-user' to 755: Operation not permitted

Note that I'm not running my container as the root user. 

Debugging the permissions on the folder, I see:

```
bash-5.1$ ls -la /nix/var/nix/profiles/         
total 12
dr-xr-xr-x 1 0 0 4096 Jan  1  1970 .
dr-xr-xr-x 1 0 0 4096 Jan  1  1970 ..
dr-xr-xr-x 1 0 0 4096 Jan  1  1970 per-user
```

I cannot manually set the permissions on this folder because whatever I do gets overridden by whatever the nix initialization code is doing. I'm not sure what the best path forward is here.

**Comment by jmgilman** [comment]
I was able to hack away at `default.nix` to get the permissions set correctly, however, it doesn't actually resolve the issue. It appears that nix doesn't care whether or not the permissions are correct, it tries to set it either way. The only context I could find [was here](https://github.com/NixOS/docker/issues/34).

**Comment by jmgilman** [comment]
Closed in favor of #50.
